src/Components/ExistingTokenButton.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import { db } from '../lib/firebase';
+import { useHistory } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+
+export default function ExistingTokenButton({ setToken }) {
+  const [input, setInput] = useState('');
+  let history = useHistory();
+  const { register } = useForm();
+
+  const handleClick = (event) => {
+    event.preventDefault();
+    if (input === '') {
+      alert('Enter a valid token');
+    } else {
+      db.collection('lists')
+        .doc(input)
+        .onSnapshot(function (querySnapshot) {
+          if (querySnapshot.exists) {
+            localStorage.setItem('token', input);
+            setToken(input);
+            history.push('/list');
+          } else {
+            alert('Shopping list does not exist. Try again or get a new token');
+          }
+        });
+    }
+  };
+
+  return (
+    <div className="token">
+      <form>
+        <input
```

Comment:
I noticed that the input here doesn't have a label element to go with it, which isn't the greatest for web accessibility. May I suggest taking the `<p>` tag that says "Join an existing shopping list by entering a three word token." and using that as a label instead, and using CSS to style the label to match the `<p>` tag? I don't believe the `label` attribute is sufficient enough and could only find information for using it with the `<track>` element. Using the placeholder text as a label would also be challenging for those with short term memories or other impairments (cognitive, visual, etc.).

https://www.w3.org/WAI/tutorials/forms/labels/
https://www.w3schools.com/tags/att_label.asp

Reply:
Thank you for catching, we just fixed!